portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java

```diff
@@ -682,6 +682,15 @@ else if (group.isLayout() || group.isLayoutSetPrototype() ||
 	}
 
 	@Override
+	public int getRegularRoleUsersGroupsCount(Role role) {
```

Comment:
I agree except for the next line, where I check the type of the role.  If you don't think it's necessary, then I might reconsider the method name altogether and go with something more generic.

Or would it make more sense to fetch the role inside this method with the roleId?